components/x-audio/src/redux/index.jsx

```diff
@@ -16,31 +16,32 @@ export default function connectPlayer (Player) {
 	const playerActions = wrapWithDispatch(store, {
 		onPlayClick: actions.requestPlay,
 		onPauseClick: actions.requestPause,
-		loadMedia: actions.loadMedia
+		loadMedia: actions.loadMedia,
+		willClose: actions.willClose
 	});
 
 	class ConnectedPlayer extends Component {
 		constructor(props) {
 			super(props);
 			this.unsubscribe = store.subscribe(this.storeUpdated.bind(this));
+			this.onCloseClick = this.onCloseClick.bind(this);
 			this.state = initialState;
 		}
 
 		componentDidMount() {
-			const { playing, url, metadata } = this.props;
-
-			playerActions.loadMedia({ url, metadata });
-
-			if (playing) 	{
-				playerActions.onPlayClick();
-			}
+			const { playing, url, trackingContext } = this.props;
+			playerActions.loadMedia({ url, trackingContext, autoplay: playing });
 		}
 
 		componentWillUnmount() {
-			playerActions.onPauseClick();
 			this.unsubscribe();
 		}
 
+		onCloseClick() {
+			playerActions.willClose();
```

Comment:
Could this go in `componentWillUnmount` maybe? Then no need to overload `onCloseClick`?